app-demo/src/main/java/co/sodalabs/demo/DemoActivity.kt

```diff
@@ -21,3 +53,34 @@ class DemoActivity : AppCompatActivity() {
         mDisposablesOnCreate.clear()
     }
 }
+
+class DemoAdapter(private val items: List<DemoViewModel>) : RecyclerView.Adapter<DemoAdapter.ViewHolder>() {
+
+    var clickListener: ((view: View, item: DemoViewModel) -> Unit)? = null
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
+        ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.demo_recycler_view_item, parent, false))
+
+    override fun getItemCount(): Int = items.size
+
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.bind(items[position])
+        holder.setOnClickListener(clickListener)
+
+    }
+
+    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+
+        private var clickListener: ((view: View, item: DemoViewModel) -> Unit)? = null
+
+        fun bind(viewModel: DemoViewModel) {
+            itemView.findViewById<TextView>(R.id.tvLabel).text = viewModel.name
```

Comment:
VERY MINOR: Just to know we could simplify this by adapting Kotlin Android extension. Could [check out the document here](https://kotlinlang.org/docs/tutorials/android-plugin.html).

Reply:
K, don't worry about it though. 👷‍♂️